src/Api/Batch/BatchRequest.php

```diff
@@ -115,13 +114,24 @@ private function processResponse(Request $request, int $status, array $body): It
     {
         // Request from batch failed, status != 2xx
         if ($status < 200 || $status >= 300) {
-            throw new BatchRequestException(sprintf(
-                'Unexpected status "%d" for request "%s": %s, %s',
-                $status,
-                $request->getUri(),
-                $body['error']['code'] ?? '',
-                $body['error']['message'] ?? '',
-            ), $status);
+            $errorCode = $body['error']['code'] ?? '';
+            switch ($errorCode) {
+                case 'GenericFileOpenError':
+                    throw new UserException('OneDrive API error: The workbook cannot be opened. Make sure ' .
+                        'nobody is editing it.');
+                case 'UnknownError':
+                    throw new UserException('OneDrive API error: The service is unavailable.');
+                case 'MaxRequestDurationExceeded':
+                    throw new UserException('OneDrive API error: Request took too long.');
+                default:
```

Comment:
Nie vsetky request sa spracovavaju ako `batch`. Mozno by bolo dobre pridat to sem:
https://github.com/keboola/wr-onedrive/blob/debbcfea7dea3a2b92042e144d44ffc31eae75e8/src/Api/Helpers.php#L72-L75

A pouzit tuto metodu aj tu. Co myslis?

Reply:
Ten typ tam mozno upravit z `RequestException` na inu, vid kde sa pouziva.

Reply:
@webrouse přesunuto